fix `incompatible-pointer-types` warnings

<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->
<!-- Please check there isn't already an issue on this topic. -->

## Environment

* SuperCollider version: latest develop (31a4bc07a16311a5ae28fd1f233167a822b048b3)
* Operating system: Windows, Msys2
* Other details (Qt version, audio driver, etc.): compiler GCC 14.1.0 (Msys2)

GCC 14 has upgraded the `-Wincompatible-pointer-types` warning to an error. As a consequence, some (external) C code in SC fails to compile, namely `portmidi` and `hidapi`:

```
C:/Repos/supercollider/external_libraries/portmidi/pm_win/pmwinmm.c:1052:31: error: assignment to 'uint32_
t *' {aka 'unsigned int *'} from incompatible pointer type 'DWORD *' {aka 'long unsigned int *'} [-Wincomp
atible-pointer-types]
 1052 |         midi->fill_offset_ptr = &(hdr->dwBytesRecorded);
      |                               ^
C:/Repos/supercollider/external_libraries/hidapi/hidapi_parser/hidapi_parser.c: In function 'hid_parse_ele
ment_info':
C:/Repos/supercollider/external_libraries/hidapi/hidapi_parser/hidapi_parser.c:1630:64: error: passing arg
ument 2 of 'HidP_GetLinkCollectionNodes' from incompatible pointer type [-Wincompatible-pointer-types]
 1630 |     nt_res = HidP_GetLinkCollectionNodes( linkCollectionNodes, &numColls, pp_data );
      |                                                                ^~~~~~~~~
      |                                                                |
      |                                                                int *
```

As a temporary fix we might pass `-Wno-error=incompatible-pointer-types` so that we still get the warning, but it's not treated as an error. Ideally, the issue should be fixed upstream.

---

Also, there is another compilation error in `SC_PAUtiles` because of a missing include of `<cstdint>`:

```
C:/Repos/supercollider/common/SC_PaUtils.cpp: In function 'PaError TryGetDefaultPaDevices(PaDeviceIndex*,
PaDeviceIndex*, int, int, double)':
C:/Repos/supercollider/common/SC_PaUtils.cpp:130:21: error: 'uint32_t' was not declared in this scope
  130 |                 if (uint32_t(inSR) != uint32_t(outSR)) {
      |                     ^~~~~~~~
C:/Repos/supercollider/common/SC_PaUtils.cpp:6:1: note: 'uint32_t' is defined in header '<cstdint>'; this
is probably fixable by adding '#include <cstdint>'
    5 | #include <cstdio>
  +++ |+#include <cstdint>
    6 |
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,10 +104,6 @@ if (${CMAKE_COMPILER_IS_GNUCXX})
 
     add_compile_options(
         -Werror=return-type      # Error out if function definitions lack a return type
-        # GCC 14 has made -Wincompatible-pointer-types an error by default and as a
-        # consequence, some external C would fail to compile. This is a temporary fix!
-        # Ideally, the affected dependencies should be fixed upstream.
-        -Wno-error=incompatible-pointer-types
         -fschedule-insns2        # Optimize instruction scheduling
         -fomit-frame-pointer     # Optimize by omitting the frame pointer where possible
         -fno-math-errno          # Optimize math functions by not setting errno

--- a/external_libraries/portmidi/pm_win/pmwinmm.c
+++ b/external_libraries/portmidi/pm_win/pmwinmm.c
@@ -1049,7 +1049,7 @@ static PmError winmm_write_byte(PmInternal *midi, unsigned char byte,
         m->hdr = hdr = get_free_output_buffer(midi);
         assert(hdr);
         midi->fill_base = (unsigned char *) m->hdr->lpData;
-        midi->fill_offset_ptr = &(hdr->dwBytesRecorded);
+        midi->fill_offset_ptr = (uint32_t *) &(hdr->dwBytesRecorded);
         /* when buffer fills, Pm_WriteSysEx will revert to calling
          * pmwin_write_byte, which expect to have space, so leave
          * one byte free for pmwin_write_byte. Leave another byte

